packages/apollo-codegen-swift/src/codeGeneration.ts

```diff
@@ -727,12 +727,12 @@ export class SwiftAPIGenerator extends SwiftGenerator<CompilerContext> {
                 expression,
                 "ResultMap",
                 structName
-              )}.filter`
+              )}?.filter`
             );
             this.withinBlock(
               () =>
                 this.printOnNewline(
-                  `${structName}.possibleTypes.contains($0.__typename)`
+                  `guard let type = $0 else { return false }; return ${structName}.possibleTypes.contains(type.__typename)`
```

Comment:
For readability it might be nice to have this on 2 lines instead.

```
guard let type = $0 else { return false };
return ${structName}.possibleTypes.contains(type.__typename)
```

Reply:
Also, I'm trying to see how `$0` can be `nil` after being `flatMap`ped.. That said, we might want to replace `flatMap` with `compactMap` since `flatMap` is deprecated.

Reply:
For some reason this was being real dumb about letting me use more than one line, and I really didn't feel like fighting with typescript at that moment. 

And I suspect it's *because* we're using `flatMap` where we should be using `compactMap`, but I'd like to hear some thoughts from @martijnwalraven before I rip this apart any further.

Reply:
(OK i at least got this onto a separate line)